javaslang/src/main/java/javaslang/collection/PriorityQueue.java

```diff
@@ -575,48 +586,31 @@ public String toString() {
        * *         val (ts',xs') = split ([],[],c)
        * * in fold insert xs' (meld (ts, ts')) end
        * */
-        protected static <T> Tuple2<List<T>, List<Node<Ranked<T>>>> deleteMinForest(SerializableComparator<? super T> comparator, List<Node<Ranked<T>>> forest) {
+        protected static <T> List<Node<Ranked<T>>> deleteMin(SerializableComparator<? super T> comparator, List<Node<Ranked<T>>> forest) {
             if (forest.isEmpty()) {
                 throw new NoSuchElementException();
             } else {
-                final Tuple2<Node<Ranked<T>>, List<Node<Ranked<T>>>> min = getMin(comparator, forest);
-                final Tuple3<List<Node<Ranked<T>>>, List<T>, ?> split = split(List.empty(), List.empty(), min._1.getChildren());
-
-                final List<T> elems = split._2;
-                final List<Node<Ranked<T>>> union = union(comparator, min._2, split._1);
-                return Tuple.of(elems, union);
-            }
-        }
-
-        private static <T> Tuple2<Node<Ranked<T>>, List<Node<Ranked<T>>>> getMin(SerializableComparator<? super T> comparator, List<Node<Ranked<T>>> forest) {
```

Comment:
after understanding what these methods -- specified in the original paper -- do, I managed to unroll and inline them, saving many stacks and `Tuple` wrappings